syntax = "proto3";

package motify.gateway;


service AccountService {
    rpc GetAccounts(GetAccountsRequest) returns (GetAccountsResponse) {}
}

service UserPlayerAuthService {
    rpc GetUserToken(GetUserTokenRequest) returns (GetUserTokenResponse) {}
}

message GetAccountsRequest {
    repeated string id = 1;
}

message GetAccountsResponse {
    map<string, Account> accounts = 1;
}

message Account {
    string id = 1;
    string email = 2;
    string username = 3;
    string date_joined = 4;
    Subscription subscription = 5;
}

message GetUserTokenRequest {
    string id = 1;
}

message GetUserTokenResponse {
    string token = 1;
    string device_id = 2;
}

enum Subscription {
    UNKNOWN = 0;
    BASIC = 1;
    STANDARD = 2;
    PREMIUM = 3;
}


